### ISSUE TITLE
Incident - reportedBy should store the currently logged in user id

### ISSUE THREAD

**Issue by blestab** [HospitalRun/hospitalrun-frontend#2539]
## 🐛 Bug Report

Report incident is currently hardcoded to set `reportedBy` to **some user** instead of the currently logged in user's id. This needs to be corrected to user the current user id from the redux store.

## To Reproduce

- Access staging https://staging.hospitalrun.io/
- Report an incident
- View the list of incidents and notice that the reported by field shows 'some user'

## Expected behavior

- The `incident.reportedBy` should be the currently logged in `user.id` at the time the incident is reported
- The list of incidents should display the `user.fullName` based on the `incident.reportedBy`

## Notes
- See `src/shared/components/navbar/Navbar.tsx` on how to access the user from the store

## Your Environment

- *node version*: 6,8,10
- *fastify version*: >=1.0.0
- *os*: Mac, Windows, Linux
- *any other relevant information*

**Comment by jameszheng405** [comment]
@blestab @jackcymeye @mormme @thekappa @fox1t Hi, this is James from Stanford Code the Change! I'd like to be assigned to this task

**Comment by morrme** [comment]
@jameszheng405  it's yours!

**Comment by blestab** [comment]
Hi @jameszheng405, 
How is this coming along? Feel free to check this PR https://github.com/HospitalRun/hospitalrun-frontend/pull/2544 for ideas

**Comment by jameszheng405** [comment]
Hi @blestab ,
I'm able to display the user id now, just working on displaying the full name in the table. I'll be meeting with Stanford CTC tomorrow to discuss this. Thanks for checking in!

**Comment by blestab** [comment]
> Hi @blestab ,
> I'm able to display the user id now, just working on displaying the full name in the table. I'll be meeting with Stanford CTC tomorrow to discuss this. Thanks for checking in!

Awesome! Thanks

**Comment by jameszheng405** [comment]
@blestab 
After taking a look at PR #2544, I noticed that fullName is statically defined and used to test the value of requestedByFullName. I was wondering if this might be inefficient in case a user decides to change their name and the value of fullName would then have to be changed everywhere? I’m thinking it would be better to always be able to access the current fullName through the user id, which never changes. However, since that information isn’t globally available yet, I would like to propose an extractFullName function which will be scaffolded later when that functionality is there.

**Comment by riiniii** [comment]
that's a good point. currently doesn't look like we have functionality to change much about user, but is good for if/when user info editing is implemented for the future.. 

following + will make changes as necessary for [#2544](https://github.com/HospitalRun/hospitalrun-frontend/pull/2544)

**Comment by sshikhar1234** [comment]
Hi is this issue fixed in production?

**Comment by tehKapa** [comment]
@sshikhar1234 which production?

**Comment by sshikhar1234** [comment]
I mean like in main release. Is this issue fixed or its open?

**Comment by tehKapa** [comment]
No, if this issue is still open, I don't think so. Why?

**Comment by akshatsinghania** [comment]
i would like to contribute fixing this issue , should i submit  a pull request?

**Comment by bottercode** [comment]
i would like to work on this one. Is it still Open?

**Comment by tehKapa** [comment]
yep @bottercode

**Comment by bottercode** [comment]
I have fixed the issue, new to this so can you just tell me if you have received the pull request i made?

**Comment by tehKapa** [comment]
> I have fixed the issue, new to this so can you just tell me if you have received the pull request i made?

@bottercode I don't see any new PR https://github.com/HospitalRun/hospitalrun-frontend/pulls. Can you link it?

**Comment by bottercode** [comment]
Sorry for the very late response here it is @tehKapa  https://github.com/bottercode/hospitalrun-frontend/commit/d327ac1e892b766f74ec6d9e76b41faad0fc433b

**Comment by tehKapa** [comment]
@bottercode you need to do a PR in hospitalrun/hospitalrun-frontend and not on your repo

**Comment by bottercode** [comment]
Is it visible now? @tehKapa

**Comment by tehKapa** [comment]
> Is it visible now? @tehKapa

No. You can see all open PR here https://github.com/HospitalRun/hospitalrun-frontend/pulls

**Comment by ykabusalah** [comment]
As this issue is currently open, has it been fixed? Or could I claim this issue and attempt it myself?

**Comment by farrukhrashid1997** [comment]
Is this issue still open? I'd like to contribute.

**Comment by xhefribala** [comment]
@tehKapa Could you please assign me to this issue?

**Comment by xhefribala** [comment]
@tehKapa Can you please review the PR https://github.com/HospitalRun/hospitalrun-frontend/pull/2987 ?

**Comment by xhefribala** [comment]
@tehKapa, can you please review the PR https://github.com/HospitalRun/hospitalrun-frontend/pull/2988
